scylla-cdc-replicator/src/replication_tests.rs

```diff
@@ -317,4 +317,62 @@ mod tests {
             .await
             .unwrap();
     }
+
+    #[tokio::test]
+    async fn test_map_update() {
+        let schema = TestTableSchema {
+            name: "MAPS_UPDATE".to_string(),
+            partition_key: vec![("pk", "int")],
+            clustering_key: vec![("ck", "int")],
+            other_columns: vec![("v1", "map<int, int>"), ("v2", "map<int, boolean>")],
+        };
+
+        let operations = vec![
+            (
+                "INSERT INTO MAPS_UPDATE (pk, ck, v1, v2) VALUES (?, ?, ?, ?)",
+                vec![
+                    Int(1),
+                    Int(2),
+                    Map(vec![(Int(1), Int(1)), (Int(2), Int(2))]),
+                    Map(vec![(Int(1), Boolean(true))]),
+                ],
+            ),
+            (
+                "INSERT INTO MAPS_UPDATE (pk, ck, v1, v2) VALUES (?, ?, ?, ?)",
+                vec![
+                    Int(10),
+                    Int(20),
+                    Map(vec![(Int(10), Int(-10))]),
+                    Map(vec![(Int(11), Boolean(false))]),
+                ],
+            ),
+            (
+                "UPDATE MAPS_UPDATE SET v1 = v1 - ? WHERE pk = ? AND ck = ?",
+                vec![Set(vec![Int(10)]), Int(10), Int(20)],
+            ),
+            (
+                "UPDATE MAPS_UPDATE SET v2 = v2 + ? WHERE pk = ? AND ck = ?",
+                vec![
+                    Map(vec![
+                        (Int(-21374134), Boolean(true)),
+                        (Int(-43142137), Boolean(false)),
+                    ]),
+                    Int(1),
+                    Int(2),
+                ],
+            ),
+            (
+                "UPDATE MAPS_UPDATE SET v2 = ? WHERE pk = ? AND ck = ?",
+                vec![Map(vec![(Int(2), Boolean(true))]), Int(10), Int(20)],
+            ),
+            (
+                "DELETE v1 FROM MAPS_UPDATE WHERE pk = ? AND ck = ?",
+                vec![Int(1), Int(2)],
+            ),
```

Comment:
Putting overwrite statements last make the test less accurate. If replicating overwriting statements works but adding/deleting elements doesn't (`v = v + ?` etc), the test will not catch it (you overwrote both columns which you previously added/deleted elements to/from). BTW. The `INSERT`s at the beginning are not necessary in this test.

You probably need multiple test cases to check every type of statement. Or maybe there is some order of statements such that every statement matters - in this test case right now, only the last two statements do.

Alternatively, we could modify `test_replication` so it compares the tables after every operation, not only at the end.

Finally, it would be great to use property based testing so there's no need to write these tedious test cases by hand.

Reply:
Now I noticed you're using different keys for the overwrites than for the additions/deletions, so I guess it's OK.